packages/components/pagination/stories/pagination.stories.ts

```diff
@@ -0,0 +1,506 @@
+} from '../src/pagination'
+import PuikPagination from './../src/pagination.vue'
+import type { Meta, Story, Args } from '@storybook/vue3'
+
+const paginationVariantsSummary = puikPaginationVariants.join('|')
+
+export default {
+  title: 'Components/Pagination',
+  component: PuikPagination,
+  argTypes: {
+    variant: {
+      control: 'select',
+      options: puikPaginationVariants,
+      description: 'Set the pagination variant',
+      table: {
+        defaultValue: {
+          summary: PuikPaginationVariantEnum.medium,
+        },
+        type: {
+          summary: paginationVariantsSummary,
+        },
+      },
+    },
+    totalItem: {
+      control: 'number',
+      description: 'Set the total item count',
+      table: {
+        type: {
+          summary: 'number',
+        },
+      },
+    },
+    itemsPerPage: {
+      control: 'none',
+      table: {
+        type: {
+          summary: 'number',
+        },
+        defaultValue: { summary: 5 },
+      },
+      description: 'v-model of the items per page',
+    },
+    itemsPerPageOptions: {
+      control: 'none',
+      description: 'Set the items per page options',
+      table: {
+        type: {
+          summary: 'number[]',
+        },
+        defaultValue: {
+          summary: puikPaginationItemsPerPageOptions,
+        },
+      },
+    },
+    page: {
+      control: 'numbnoneer',
+      description: 'v-model of the current page',
+      table: {
+        type: {
+          summary: 'number',
+        },
+      },
+    },
+    itemCount: {
+      control: 'number',
+      description: 'Set the current item count',
+      table: {
+        type: {
+          summary: 'number',
+        },
+        defaultValue: {
+          summary: '0',
+        },
+      },
+    },
+    label: {
+      control: 'text',
+      description: 'Set the label',
+      table: {
+        defaultValue: {
+          summary: 'undefined',
+        },
+        type: {
+          summary: 'string',
+        },
+      },
+    },
+    loaderButtonLabel: {
+      control: 'text',
+      description: 'Set the text used in button on loader variant',
+      table: {
+        defaultValue: {
+          summary: 'undefined',
+        },
+        type: {
+          summary: 'string',
+        },
+      },
+    },
+  },
+  args: {
+    variant: PuikPaginationVariantEnum.medium,
+    totalItem: 500,
+    itemCount: 25,
+    label: '',
+    loaderButtonLabel: undefined,
+  },
+} as Meta
+
+const Template: Story = (args: Args) => ({
+  components: {
+    PuikPagination,
+  },
+  setup() {
+    const page = ref(1)
+    const itemsPerPage = ref(5)
+    return { args, page, itemsPerPage }
+  },
+  template: `<puik-pagination
+      v-model:page="page"
+      v-model:items-per-page="itemsPerPage"
+      v-bind="args"
+    ></puik-pagination>`,
+})
+
+export const Default = Template.bind({})
+Default.args = {}
+Default.parameters = {
+  docs: {
+    source: {
+      code: `
+      <!--VueJS Snippet-->
```

Comment:
You need to remove tabulation here to prevent wrong formatting in the Storybook documentation